tests/test_estimate_params.py

```diff
@@ -905,6 +905,188 @@ def test_incorrect_lens(self):
+
+        # These cases are working?
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [])
+        hold1 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+
+        # Tolerance works as intended as long as it is within a sequence of length 1
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [15])
+        hold2 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        # self.assertEqual(hold2, check2)
+        self.assertNotEqual(hold1, hold2)
+
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = 15)
+        hold1 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        self.assertEqual(hold1, hold2)
+
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = (15))
+        hold2 = m.calc_error(results.times, results.inputs, results.outputs)
+
+        self.assertEqual(hold1, hold2)
+
+        # Cannot pass Sets into tolerance
+        with self.assertRaises(TypeError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = {1})
+
+        # Works when tolerance is a floating point value.
+
+        m.parameters['thrower_height'] = 3.1
+        m.parameters['throwing_speed'] = 29
+        m.parameters['g'] = 10
+        m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = 1.123543297)
+        if not all(abs(m.parameters[key] - gt[key]) > 0.02 for key in keys):
+            self.fail("m.parameter shouldn't be too close to the original parameters")
+
+        # When tolerance is 
+        with self.assertRaises(ValueError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [1, 2, 3])
+
+        # 
+        with self.assertRaises(TypeError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = [1, '2', '3'])
+        
+        with self.assertRaises(ValueError):
+            m.estimate_params(times = results.times, inputs = results.inputs, outputs = results.outputs, keys = keys, tol = (1, 2, 3))
+        
+        # Using TNC
+        m.parameters['thrower_height'] = 3.1
```

Comment:
The tests on lines 1019 - 1043 could use some additional comments to describe what you are testing. This will be helpful to make sure that each test we include is doing something different and worthwhile